pkg/env/env.go

```diff
@@ -116,6 +117,21 @@ func newTestEnvWithParallel() *testEnv {
 	}
 }
 
+type ctxName string
+
+// newChildTestEnv returns a child testEnv based on the one passed as an argument.
+// The child env inherits the context and actions from the parent and
+// creates a deep copy of the config so that it can be mutated without
+// affecting the parent's.
+func newChildTestEnv(e *testEnv) *testEnv {
+	childCtx := context.WithValue(e.ctx, ctxName("parent"), fmt.Sprintf("%s", e.ctx))
```

Comment:
Why turn the parent context into a string of values here ? Since the parent context is not of much use, can this be replaced with `context.WithoutCancel(e.ctx)` instead ?

Reply:
The reason why I used this was because:
1) we need to create a new context from the parent's
2) we still want to propagate cancel from the parent to the child (I don't see why we would want to not do that)

Because of 2), we can't use `context.WithoutCancel`.
Which only leave `context.WithValue` and `context.WithCancel`:
- `WithCancel`: IIRC `go vet` would not let me ignore the cancel func with `_` 
- `WithValue` was not perfect because we do not need any value to be passed down but I couldn't find better

Happy to change it if you consider that we can ignore the cancellation from the parent's context though. What do you think?

Reply:
@Fricounet Good point. I see what you mean. 

The only reason I brought up this question was because we are converting the context into a string and setting it as value. Which is almost of no-use. Can we set something else as the value instead ?